include/flatbuffers/flatbuffers.h

```diff
@@ -515,6 +509,27 @@ class DetachedBuffer {
   size_t reserved_;
   uint8_t *cur_;
   size_t size_;
+
+  inline void destroy() {
+    if (buf_) {
+      assert(allocator_);
+      allocator_->deallocate(buf_, reserved_);
+    }
+    if (own_allocator_ && allocator_) {
+      delete allocator_;
+    }
+
+    reset();
+  }
+
+  inline void reset() {
```

Comment:
I'll re-post my comment here in case it gets lost on the outdated diff: at least for `std::unique_ptr<T>`, `reset` destroys the previous object, vs `release` just detaches it so someone else can manage it. I don't have a strong preference since `reset` can also be taken to mean "reset my own internal state". For what it's worth, there are other funcs in flatbuffers with release-related names.